Tumbleweed icons&colors changes

ilvipero opened the pull request:
1. air-openSUSE. Left it as it was, using oxygen icons. Very good theme. we might want to switch to the icons i customized for openSUSElight, darker icons might look better with air-openSUSE. but can do that later.
2. openSUSE. I changed it to have a green start icon, dark bars with green accents, oxygen icons fallback, left transitions as they were in openSUSElight and openSUSEdark.
3. openSUSElight. I removed the icons as the icon pack was not complete. mixed with breeze did not have a very good feel to it, in my opinion. So i modified all oxygen icons, inverted the colors so that it suits well on a light theme. (I will also submit oxygen-light icons to KDE and see if they like it).
4. openSUSEdark. I removed the icons as the icon pack was not complete. mixed with breeze did not have a very good feel to it, in my opinion. replaced with oxygen icons fallback.

Vogtinator commented:
Changing the start icon would lose the adaption to color scheme changes, so I'm against that.

In the theme metadata, the `X-KDE-PluginInfo-Version` needs to be bumped as otherwise the cache is not updated.

Vogtinator commented:
The openSUSE theme needs to be close to upstream, so it has to use Breeze as fallback (which is the default).

ilvipero commented:
Regarding green icon on opensuse theme: Sure, I thought to change it a bit to avoid the theme being too similar to openSUSEdark. But if you are not for it, I can revert back.

Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?

lbeltrame commented:
Some screenshots would be nice.

ilvipero commented:
> Some screenshots would be nice.

Sure, Please check attached screenshots. I am also attaching all 3 themes packaged. I renamed  both the archives and theme names in metadata files, so that you guys can import manually and they won't conflict with existing opensuse themes.
I hope it helps.

![opensuse-theme-screenshot](https://user-images.githubusercontent.com/6520987/57092025-0ffa3800-6d3d-11e9-9d54-eab70055cce4.png)
![opensusedark-theme-screenshot](https://user-images.githubusercontent.com/6520987/57092038-17b9dc80-6d3d-11e9-89a1-bf7c2f9a7233.png)
![opensuselight-theme-screenshot](https://user-images.githubusercontent.com/6520987/57092051-1c7e9080-6d3d-11e9-8eff-bda9ddd50826.png)
[openSUSE-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139037/openSUSE-workinprogress.tar.gz)
[openSUSEdark-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139040/openSUSEdark-workinprogress.tar.gz)
[openSUSElight-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139048/openSUSElight-workinprogress.tar.gz)

Vogtinator commented:
> Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?

I'd say yes, unless it looks bad.

ilvipero commented:
> > Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?
> 
> I'd say yes, unless it looks bad.

No, it does not look bad at all. I am going to make the changes.

ilvipero commented:
> Changing the start icon would lose the adaption to color scheme changes, so I'm against that.

I tried looking at the manual to see how I can keep same icon and have it adapt to color schemes, but did not find which configuration does that. Could you point me in the right direction? Happy to learn and change per your comment.

Vogtinator commented:
> No, it does not look bad at all. I am going to make the changes.

Thanks!

> > Changing the start icon would lose the adaption to color scheme changes, so I'm against that.
> 
> I tried looking at the manual to see how I can keep same icon and have it adapt to color schemes, but did not find which configuration does that. Could you point me in the right direction? Happy to learn and change per your comment.

You need to specify special CSS style sheets with a special id in the .svg file: https://techbase.kde.org/Development/Tutorials/Plasma5/ThemeDetails#Using_system_colors

ilvipero commented:
I completed all the changes per your request. I decided to leave the original start icon on all themes, it looks quite good to me. 
Could you please check and let me know if further changes are needed? 

Thanks!

Vogtinator commented:
Can you do another round of screenshots please?

ilvipero commented:
openSUSE Dark Theme. This one had mismatching icons and some missing ones. I removed all the custom icons, kept animations, colors as they were. and defaulted to breeze-dark.

![opensuse-dark-screenshot01](https://user-images.githubusercontent.com/6520987/60649406-80742100-9e74-11e9-9793-6e640cd29973.png)

ilvipero commented:
openSUSE Light Theme. This one was fine but icons were not upstream and looked a bit too different from other themes. I removed all the custom icons, kept animations, colors as they were. and defaulted to breeze-light.

![opensuse-light-screenshot01](https://user-images.githubusercontent.com/6520987/60649574-d34dd880-9e74-11e9-9864-45309dd2e33d.png)

ilvipero commented:
openSUSE Theme. I made this theme a bit more green. defaulted icons to breeze-dark.
I would have liked a green start logo to match the green openSUSE theme. But dynamically assigned colors did not look good. So for now i reverted back to the black and white icon. 

![opensuse-screenshot01](https://user-images.githubusercontent.com/6520987/60649596-e06ac780-9e74-11e9-9ce2-1f1f4f5ad5d0.png)

The icons below would look good on this theme, at least in my humble opinion. What do you think? do you guys have something green we can use as the main openSUSE theme?

![openSUSE-start](https://user-images.githubusercontent.com/6520987/60650023-dbf2de80-9e75-11e9-80b3-0e4ef5d6f894.png)

Vogtinator commented:
I'd like to have input from @hellcp here as well.

Do you have any .svgz copied from the theme set as fallback here? If so, those could be deleted. AFAICT the diff between the upstream theme set as fallback and the downstream result could be much smaller.

Also note that this seems to conflict with #14 by @guoyunhe, so you need to coordinate the changes.

ilvipero commented:
I removed all icons that come from breeze theme, and set breeze as fallback. so that the theme is smaller.

Regarding conflict, that seems related to background colors, I am happy to work with @guoyunhe

guoyunhe commented:
Hi @ilvipero , my PR is only for the color change and it is small. You can download the patch and merge it into yours (probably modify if here are things wrong in my patch). Then I can close my PR and only merge this one. Cheers.

My patches:

[patches.zip](https://github.com/openSUSE/plasma-opensuse/files/3381189/patches.zip)

ilvipero commented:
@guoyunhe Thank you, I am happy to merge your patch in my pull request. 

謝謝你 :)

ilvipero commented:
Sorry for slow reply. It seems my last comment was gone. I completed all changes from request #14 a while back and my changes are done.

DanielNoga commented:
I like that icons are removed. For example I didn't know until now, why my VPN plasma-nm icon was almost invisible: https://bugzilla.opensuse.org/show_bug.cgi?id=1134234

Can it be backported to Leap 15.1 after merge?

ilvipero commented:
@Vogtinator could you please check and if no further changes are needed, merge the request?
Thanks!

ilvipero commented:
@DanielNoga this is the bug I opened for the same issue. I will mark it resolved if the pull requests is merged. https://bugzilla.opensuse.org/show_bug.cgi?id=1128854

Vogtinator commented:
> Can it be backported to Leap 15.1 after merge?

This whole PR definitely not, it's too big and invasive (it changes the default theme!). For 15.2 it's doable though.

Vogtinator commented on config-files/usr/share/plasma/desktoptheme/openSUSElight/metadata.desktop:
```diff
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=openSUSElight
-Comment=Default openSUSE theme
+Comment=light openSUSE theme
```
`Light`

ilvipero replied:
Yes you are right. I removed all icons so that themes can inherit from upstream (breeze-light and breeze-dark). 
The svgz I added back:
opensuse start.sgvz , in the icons folder. 
indicators.svgz , in the icontasks folder. This one was already in opensuse themes, it looks good so I believe we can keep it. If you prefer to remove it, please let me know.

![start-svgz](https://user-images.githubusercontent.com/6520987/68827915-a3de9080-06de-11ea-96b5-27c4a4f1b3db.png)
![indicators-svgz](https://user-images.githubusercontent.com/6520987/68827919-aa6d0800-06de-11ea-99ed-09da479a360f.png)

Vogtinator commented on config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config:
```diff
@@ -0,0 +1,157 @@
+[Group-Default]
```
AFAIK lancelot is dead, so this could be removed.

ilvipero replied:
Shall I remove the whole lancelot folder and its contents from all themes?

Vogtinator replied:
Instead of adding files, yes. Keeping the old stuff around is IMO fine.

ilvipero commented:
> So you removed many icons, but also added some new .svgz files. Where are they from? IMO the default openSUSE theme should use upstream icons wherever possible.
> 
> If they're just copied from breeze/oxygen/etc., please let the theme inherit them instead.

Yes I removed most icons and only left the default opensuse start.svgz and indicators.svgz
indicators.svgz was previously in opensuse themes, I did not create it, just kept it. it looks good to me and we can keep it in my opinion. But if you want the whole theme to be upstream breeze, I can remove it as well.

![start-svgz](https://user-images.githubusercontent.com/6520987/68831444-56b3ec00-06e9-11ea-99e6-514efdcc6776.png)
![indicators-svgz](https://user-images.githubusercontent.com/6520987/68831445-59164600-06e9-11ea-8f81-06f608bb07ef.png)

noahadvs commented:
One thing I would like to see either now or in the future is the use of SVG in the repo instead of SVGZ. Upstream Breeze switched to using SVG in the repo and Git isn't able to read the contents of SVGZ files, so they cause lots of repo history bloat and make patch review more difficult. When the upstream Breeze desktop theme is built, the SVG files are converted to SVGZ.

Vogtinator commented on config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" standalone="no"?>
```
Is a plain SVG file, but with `svgz` extension.

ilvipero replied:
The file was renamed per your request.

ilvipero replied:
So, originally I did not want to deviate too much from the themes that are currently on opensuse tumbleweed. Most of them are heavily customized compared to breeze variants. 
I thought to just tweak the minimum to get it improved. But if you prefer to have a major overhaul of those and bring opensuse themes more in line with breeze, I am happy to help and will do my best.

Vogtinator replied:
> I thought to just tweak the minimum to get it improved. But if you prefer to have a major overhaul of those and bring opensuse themes more in line with breeze, I am happy to help and will do my best.

If the theme is meant to be enabled by default, this is IMO a necessary precondition. For non-default themes it doesn't matter as much.

ilvipero replied:
I have no issue at all with that. I did not create those themes so I did not want to butcher other people's work too much :)
So if I remember well we already fallback to breeze dark or breeze light on those 3 themes. Means if no item is found, fallback theme is used instead. Does it mean we can basically remove most of the icons and widgets and only leave the branding and start icons?

Vogtinator replied:
The branding and start icons are already injected into the breeze and oxygen icon themes (`/usr/share/icons/breeze/places/16/start-here-branding.svg`) so not part of the theme or lnf.

hellcp replied:
Maybe we could also adjust the highlight/accent colours, considering Plasma guys want to make that easier for the users anyway https://phabricator.kde.org/T11052

hellcp replied:
And I mean breeze green, not openSUSE green, openSUSE green really doesn't work well with breeze

ilvipero replied:
@Vogtinator @hellcp  I made all the modifications, left a very small set of files for color schemes, metadata and start icon in svg format. I will upload a few screenshots and from those you can also see the themes folder structure is very clean. Please let me know if it looks ok.
@guoyunhe your patch was not removed. Your color changes still apply.

ilvipero replied:
openSUSE theme.
![openSUSE-breezedefaults](https://user-images.githubusercontent.com/6520987/74707239-3188c980-5254-11ea-9f88-04970f97e78d.jpg)

openSUSE dark theme.
![openSUSEdark-breezedefaults](https://user-images.githubusercontent.com/6520987/74707272-4ebd9800-5254-11ea-9605-1c5deec9e3dc.jpg)

openSUSE light theme.
![openSUSElight-breezedefaults](https://user-images.githubusercontent.com/6520987/74707296-5c731d80-5254-11ea-9fce-b852eb2f81b3.jpg)

Vogtinator commented:
Looking at the diff (not the commits) there are still many `.svg` files getting added, which is a maintenance burden.
Except for openSUSE-specific SVGs, everything else has to be inherited from the upstream theme.
The old themes didn't do that, which is the reason why they're unmaintained/incomplete now.

IMO though `openSUSE` should stay as a light theme as it is right now, but we can switch the default desktoptheme to `openSUSEdark`. Otherwise users of the default `openSUSE` theme would suddenly have a dark shell, which isn't really expected.

Theme  | Inherits from | Changes to the inherited theme
------------ | ------------- | ---
openSUSE | Breeze | `start.svgz`, `indicators.svgz` and maybe colors (?)
openSUSE Dark | Breeze Dark | `start.svgz`, `indicators.svgz` and colors
openSUSE Light | Breeze Light | `start.svgz`, `indicators.svgz` and colors (?)

The screenshots look good and I'm also in favor of switching to a dark theme by default, but the majority of `.svg` files here are AFAICT not necessary and should be deleted. Essentially, what should be done is to create a new theme folders from scratch and only copy over necessary files.

ilvipero commented:
@Vogtinator Thanks for the insight. I will make brand new clean folders and just leave the very few files for color scheme and start.svgz.

ilvipero commented:
@Vogtinator 
Per you recommendation, I cleaned up unnecessary .svg(z) to bring themes close to upstream.
FYI, there is one theme, openSUSEdefault. It takes the fallback from the air theme, not breeze. I did not know what you wanted to do with it, so I left it there.

DanielNoga commented:
Hello, what is status here? It would be good to fix missing icons in dark theme by merging this patch

ilvipero commented:
> Hello, what is status here? It would be good to fix missing icons in dark theme by merging this patch

Hi, I have done all the changes that were asked, a long time ago, and was waiting for more feedback or confirmation it could be merged. @Vogtinator

Vogtinator commented:
I'll merge this now (squashing everything) for the Plasma 5.24 submission to Tumbleweed, and fix some minor issues I noticed while testing this.

However, I'll revert some color changes for now:

Switching the default to a dark theme is a rather invasive change, as it hits existing installations. On top of the change itself, it appears like some Plasma caches aren't invalidated and after installation there's an unnusable mixup of old and new colors being used, leading to e.g. unreadable text. (It might be possible to address some of this by switching the LNF to use `openSUSEdark`, instead of making `openSUSE` a dark theme, but I haven't tried yet)

The openSUSE color scheme results in green folder icons, and the contrast between the green background and the symbols on top is rather poor:

![Screenshot_20220116_215519](https://user-images.githubusercontent.com/1622084/149677734-755fc367-e00d-46ee-a04f-8379adf4e865.png)

We can revert the reverts once the issues got addressed.
